### PR TITLE
Improve export 

### PR DESCRIPTION
**Pull request by naudinlo**
## Fixes
- Remove `set` from export: keep multiple values as `list` when they come from a resource other than Patient, else keep only one (to avoid multiple similar values, ex. Name)
- Remove `Patient:from_id` column from export, representing patient index in FHIR
- Fix wrong ICD-9 display

### CODE REVIEW

File: jpyltime/app.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def fhir2dataset_route(
 
     # postprocessing
     df = FHIR2DS_Postprocessing(updated_map_attributes).postprocess(sql_df, requested_attributes)
+    print(df)
```
Comment (Jasopaum) [inline jpyltime/app.py:59]: Do you really want to keep this print?

File: jpyltime/postprocessing_fhir2ds.py [hunk 2]
```diff
@@ -87,7 +112,9 @@ def postprocess(
         Returns:
             pd.DataFrame: The dataFrame updated with the previous transformation.
         """
-        df = self._concatenate_columns(df, attributes, patient_index)
-        df = self._groupby_one_column(df, col_for_merging=patient_index)
+        df, col_from_patient_table = self._concatenate_columns(df, attributes, patient_index)
+        df = self._groupby_one_column(
+            df, col_for_merging=patient_index, col_from_patient_table=col_from_patient_table
+        )
```
Comment (Jasopaum) [inline jpyltime/postprocessing_fhir2ds.py]: Couldn't we group by all the columns that come from the Patient table instead?

File: jpyltime/app.py (hunk 1 above)
Comment (naudinlo) [reply jpyltime/app.py:59]: Yes, to easy the debug when looking at the log

File: jpyltime/postprocessing_fhir2ds.py (hunk 2 above)
Comment (naudinlo) [reply jpyltime/postprocessing_fhir2ds.py]: Good point

File: jpyltime/postprocessing_fhir2ds.py [hunk 3]
```diff
@@ -12,36 +12,61 @@ def __init__(self, map_attributes: Dict[str, Any], anonymization_symbol: str = "
         self.map_attributes = map_attributes
         self.anonymization_symbol = anonymization_symbol
 
-    def _groupby_one_column(self, df: pd.DataFrame, col_for_merging: str) -> pd.DataFrame:
-        """Groupby one column (arg. col_for_merging) and combine the grouped rows in list, keeping only not null value
-        Check that the column selected to apply the groupby exists in the dataFrame, overwise raise an Exception."""
+    def _groupby_one_column(
```
Comment (Jasopaum) [inline jpyltime/postprocessing_fhir2ds.py]: You can rename to `_groupby_patient_column`
And I also think that you could give directly all the columns as argument instead of computing them in the function

File: jpyltime/postprocessing_fhir2ds.py [hunk 4]
```diff
@@ -87,7 +116,10 @@ def postprocess(
         Returns:
             pd.DataFrame: The dataFrame updated with the previous transformation.
         """
-        df = self._concatenate_columns(df, attributes, patient_index)
-        df = self._groupby_one_column(df, col_for_merging=patient_index)
+        df, col_from_patient_table = self._concatenate_columns(df, attributes, patient_index)
```
Comment (Jasopaum) [inline jpyltime/postprocessing_fhir2ds.py]: The fact that you compute `col_from_patient_table` in a function that's meant to do something else is a bit hard to follow
I think I would have done something like 
```
col_from_patient_table = {attribute.custom_name for atribute in attributes.values() if self.map_attributes[attribute.official_name].get("fhir_resource") == patient_resource_name}
```
outside of the function

Comment (Jasopaum) [reply jpyltime/postprocessing_fhir2ds.py]: And then give it to the groupby function

File: jpyltime/postprocessing_fhir2ds.py [hunk 5]
```diff
@@ -87,7 +116,10 @@ def postprocess(
...
+        df = self._groupby_one_column(
+            df, col_for_merging=patient_index, col_from_patient_table=col_from_patient_table
+        )
         df = self.anonymize(df, attributes)
+        df.drop(columns=[patient_index], inplace=True)
```
Comment (Jasopaum) [inline jpyltime/postprocessing_fhir2ds.py:107]: Maybe add a comment to tell that we don't want to keep the col with the id (by the way, do we still need to fetch it?)

File: jpyltime/app.py (hunk 1 above)
Comment (Jasopaum) [reply jpyltime/app.py:59]: 👍

File: jpyltime/postprocessing_fhir2ds.py [hunk 6]
```diff
@@ -12,36 +12,61 @@ def __init__(self, map_attributes: Dict[str, Any], anonymization_symbol: str = "
...
-        return df.groupby(by=[col_for_merging]).agg(lambda x: set(x[x.notna()])).reset_index()
+        groupby_cols = [col_for_merging] + list(
+            [
+                col
+                for col, is_from_patient_table in col_from_patient_table.items()
+                if is_from_patient_table
+            ]
+        )
+        return (
+            df.groupby(groupby_cols, dropna=False).agg(lambda x: list(x[x.notna()])).reset_index()
```
Comment (Jasopaum) [inline jpyltime/postprocessing_fhir2ds.py]: I tried a few times but I still don't understand `lambda x: list(x[x.notna()])` 😅 
Could you add a comment to explain it? Maybe it'll help the next person who will work on this 🙂

File: jpyltime/postprocessing_fhir2ds.py [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import json
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, List, Optional, Tuple
```
Comment (Jasopaum) [inline jpyltime/postprocessing_fhir2ds.py]: I don't think you use `Tuple` anymore

File: jpyltime/postprocessing_fhir2ds.py [hunk 8]
```diff
@@ -83,11 +82,29 @@ def postprocess(
...
         df = self._concatenate_columns(df, attributes, patient_index)
-        df = self._groupby_one_column(df, col_for_merging=patient_index)
+        # Construct the list of columns coming from the main table, Patient"""
+        patient_columns: List[str] = [patient_index]
+        for attribute in attributes.values():
+            if (
+                self.map_attributes[attribute.official_name]["fhir_resource"]
+                == patient_resource_name
+            ):
+                patient_columns.append(attribute.custom_name)
```
Comment (Jasopaum) [inline jpyltime/postprocessing_fhir2ds.py]: I would have written it with a list comprehension (because I love them 😍) but if you prefer your version, you can leave it as is
```
# Maybe we can even drop the patient_index col here
patient_columns: List[str] = [
    attr.custom_name for attr in attributes.values()
    if self.map_attributes[attribute.official_name]["fhir_resource"] == patient_resource_name
]
```

**Review by Jasopaum — APPROVED** [review]
A few nit-picky comments but well done 🚀
